rotkehlchen/api/v1/schemas.py

```diff
@@ -2575,6 +2575,40 @@ def transform_data(
             blockchain=data['blockchain'],
         )
 
+    @validates_schema
+    def validate_schema(
+            self,
+            data: dict[str, Any],
+            **_kwargs: Any,
+    ) -> None:
+        if data['blockchain'] is None:
+            return
+
+        blockchain = cast(SupportedBlockchain, data['blockchain'])
+        supported_blockchain_type = blockchain.get_chain_type()
+        if supported_blockchain_type == 'evm':
+            try:
+                to_checksum_address(data['address'])
+            except (ValueError, TypeError) as e:
+                raise ValidationError(
+                    f'Given value {data["address"]} is not an {blockchain} address',
+                    field_name='address',
+                ) from e
+        elif (
+            (
+                supported_blockchain_type == 'bitcoin' and
+                is_valid_btc_address(data['address']) is False
+            ) or
+            (
+                supported_blockchain_type == 'substrate' and
+                is_valid_substrate_address(chain=blockchain, value=data['address']) is False  # type: ignore  # expects polkadot or kusama
+            )
+        ):
+            raise ValidationError(
+                f'Given value {data["address"]} is not a {blockchain} address',
+                field_name='address',
+            )
+
```

Comment:
This approach has a serious flaw which you should also test for.

You are doing `to_checksum_address` in the validation. It should happen in `post_load` and save the new value there.

The reason for this is that if somehow a non-checksummed but valid EVM address is given to the API then it will happilly pass this validation, but what will end up getting propagated further down is the non-checksummed address.